src/System.Private.CoreLib/src/System/Reflection/RuntimePropertyInfo.cs

```diff
@@ -224,6 +224,7 @@ private RuntimeType ReflectedTypeInternal
 
         public override Module Module { get { return GetRuntimeModule(); } }
         internal RuntimeModule GetRuntimeModule() { return m_declaringType.GetRuntimeModule(); }
+        public override bool IsCollectible => m_getterMethod.IsCollectible || m_setterMethod.IsCollectible;
```

Comment:
Actually, I do not think you can have generic properties. So it means it should be fine to just return collectibility of the type like we are doing it for fields.

Reply:
Well, you can't have generic properties in C# today, but I don't see any reason from the ECMA standard why you couldn't. I would prefer to you see you keep the current implementation.